src/java/org/apache/cassandra/service/accord/CommandsForRanges.java

```diff
@@ -155,16 +202,93 @@ public void onUpdate(AccordCacheEntry<TxnId, Command> state)
                     {
                         RangeRoute cur = cachedRangeTxnsById.put(cmd.txnId(), upd);
                         if (!upd.equals(cur))
-                        {
-                            if (cur != null)
-                                remove(txnId, cur);
-                            cachedRangeTxnsByRange = IntervalBTree.update(cachedRangeTxnsByRange, toMap(txnId, upd), COMPARATORS);
-                        }
+                            IntervalTreeEdit.push(new IntervalTreeEdit(txnId, toMap(txnId, upd), cur == null ? null : toMap(txnId, cur)), this);
                     }
                 }
             }
         }
 
+        private void pushEdit(IntervalTreeEdit edit)
```

Comment:
this is private and nothing touches it; is this dead code?

Reply:
oh this must be a bad rebase, hopefully nothing else lost. let me fix that now